src/components/inputs/text/NameTokenSelector/NameTokenSelector.tsx

```
@@ -30,85 +47,281 @@ function NameTokenSelector({
+      if (imports) {
+        const details = newTokens[imports[0].toString()];
+        setSelectedToken({
+          name: details?.name,
+          ticker: details?.ticker,
+          id: imports[0].toString(),
+        });
+      }
+    } catch (error: any) {
+      eventEmitter.emit('error', error);
+    } finally {
+      setLoading(false);
+      setListPage(1);
+    }
+  }
+
+  function handleTokenSearch(query: string) {
     try {
       setSearching(true);
-      if (!id.length) {
+      setValidImport(undefined);
+      if (!query) {
         setSearchText('');
+        setSelectedToken(undefined);
         return;
       }
+      setSearchText(query);
 
-      setSearchText(id);
-      const txid = new ArweaveTransactionID(id);
-      selectedTokenCallback(txid);
-      const state =
-        await arweaveDataProvider.getContractState<PDNTContractJSON>(txid);
-      const contract = new PDNTContract(state, txid);
-
-      if (!contract.isValid()) {
-        throw Error('Contract does not match required schema');
+      if (!tokens) {
+        throw new Error('No Name Tokens Found');
       }
-      setSelectedNameToken({
-        id,
-        name: contract.name,
-        ticker: contract.ticker,
-      });
-      setValidImport(true);
-      setSearchText('');
+      const filteredResults = Object.keys(tokens)
+        .filter((id) => {
+          const { owner, controller, name, ticker } = tokens[id];
+          const queryResult = [owner, controller, name, ticker, id].some(
+            (term) =>
+              term &&
+              term.toString().toLowerCase().includes(query.toLowerCase()),
+          );
+          return queryResult;
+        })
+        .map((id) => {
+          const { name, ticker } = tokens[id];
+          return { id, name: name ?? '', ticker: ticker ?? '' };
+        })
+        .filter((n) => n !== undefined);
+      if (!filteredResults.length) {
+        throw new Error('No ANT tokens found for that search');
```

do we actually want to throw an error here?

---

For the moment yes.